script.js

```diff
@@ -1 +1,341 @@
+      };
+
+      // console.log(`rank: ${rankCounter}`);
+
+      // add the card to the deck
+      newDeck.push(card); // add double the cards to the deck
+      newDeck.push(card);
+    }
+  }
+
+  return newDeck;
+};
+// randomIndex for shuffle cards
+const getRandomIndex = (size) => Math.floor(Math.random() * size);
+const shuffleCards = (cards) => {
+  // loop over the entire cards array
+  for (let currentIndex = 0; currentIndex < cards.length; currentIndex += 1) {
+    // select a random position from the deck
+    const randomIndex = getRandomIndex(cards.length);
+
+    // get the current card in the loop
+    const currentItem = cards[currentIndex];
+
+    // get the random card
+    const randomItem = cards[randomIndex];
+
+    // swap the current card and the random card
+    cards[currentIndex] = randomItem;
+    cards[randomIndex] = currentItem;
+  }
+
+  // return shuffled deck
+  return cards;
+};
+
+// Starting instructions
+const gameInstructionDiv = document.createElement('div');
+gameInstructionDiv.setAttribute('class', 'display');
+// Output for Instructions
+const gameMessage = (message) => {
+  gameInstructionDiv.innerHTML = message;
+};
+
+// creating display Message that disappear after 3 seconds
+const messageDisplay = (boardElement) => {
+  const messageDisplayDiv = document.createElement('div');
+  messageDisplayDiv.classList.add('message-display-div, display');
+  messageDisplayDiv.innertext = 'Its a match!';
+  boardElement.appendChild(messageDisplayDiv);
+
+  // prevent user from having the ability to click
+  abilityToClick = false;
+  // using setTimeOut() to make message disappear after 3 seconds
+  setTimeout(() => {
+    boardElement.removeChild(messageDisplayDiv);
+    abilityToClick = true;
+    // change game message
+    gameMessage('Please wait 3 Seconds...<br> Please press another card to match!');
+  }, 3000);
+};
+// No match by user
+const noMatch = (firstSquare, secondSquare) => {
+  abilityToClick = false;
+  setTimeout(() => {
+    firstSquare.innerText = '';
+    secondSquare.innerText = '';
+    abilityToClick = true;
+  }, 3000);
+};
+
+// Timer display
+let timerID = 0;
+const countDownEl = document.createElement('text');
+countDownEl.setAttribute('class', 'display');
+let currentTime = 0;
+// countDown function
+const countDown = () => {
+  currentTime -= 1;
+  countDownEl.innerHTML = `Time Remaining: ${currentTime}`;
+  console.log(currentTime);
+  if (currentTime === 0) {
+    clearInterval(timerID);
+    // eslint-disable-next-line no-alert
+    alert(`Game Over! Your total matched is ${totalMatched}`);
+    // eslint-disable-next-line no-restricted-globals
+    location.reload();
+  }
+};
+// change matched display
+const changeMatchedDisplay = (firstSquare, secondSquare) => {
+  firstSquare.innerHTML = '✅';
+  secondSquare.innerHTML = '✅';
+};
+// Score display
+const scoreMessageDiv = document.createElement('div');
+scoreMessageDiv.setAttribute('class', 'display');
```

Comment:
Careful putting this code that runs on page load in between functions that only run when called. I would put all the initialisation in the `gameInit` function, and if you need to declare globals at the highest scope, declare all of them at the top so that you can see all of them in 1 place.

E.g. at the top of the file you might have:
```const scoreMessageDiv;```

and in `gameInit` you might have:
```
scoreMessageDiv = document.createElement('div');
...
```